pkg/compactor/blocks_cleaner.go

```diff
@@ -513,13 +513,13 @@ func (c *BlocksCleaner) cleanUserPartialBlocks(ctx context.Context, partials map
 	// Check if partial blocks are older than delay period, and mark for deletion
 	if !partialDeletionCutoffTime.IsZero() {
 		for _, blockID := range partialBlocksWithoutDeletionMarker {
-			lastModified, err := findMostRecentModifiedTimeForBlock(ctx, blockID, userBucket)
+			outside, err := isPartialBlockOutsideDelayPeriod(ctx, blockID, userBucket, partialDeletionCutoffTime)
 			if err != nil {
-				level.Warn(userLogger).Log("msg", "failed to find last modified time for partial block", "block", blockID, "err", err)
+				level.Warn(userLogger).Log("msg", "failed while determining if partial block should be marked for deletion", "block", blockID, "err", err)
 				continue
 			}
-			if !lastModified.IsZero() && lastModified.Before(partialDeletionCutoffTime) {
-				level.Info(userLogger).Log("msg", "stale partial block found: marking block for deletion", "block", blockID, "last modified", lastModified)
+			if outside {
+				level.Info(userLogger).Log("msg", "stale partial block found: marking block for deletion", "block", blockID)
```

Comment:
Logging "last modified" is useful to confirm that we're deleting the right thing. It would be nice to keep it.

Reply:
Sure, I can change the function signature back to preserve this. Maybe it would be useful to log the deletion delay here as well.

Reply:
Brought back the original log message without changes.

Reply:
Thank you!